src/common/components/UserProfileAvatar.tsx

```diff
@@ -0,0 +1,20 @@
+import { Avatar, type AvatarProps } from "@radix-ui/themes";
+
+type UserProfileAvatarProps = Omit<
+  AvatarProps,
+  "radius" | "variant" | "color" | "fallback"
+>;
```

Comment:
연서언니가 달아준 제 [pr 코리 내용](https://github.com/Mo-yoy/Moyo-Frontend/pull/14#discussion_r1921970894)에서의 이점과 결론처럼 interface 사용하는건 어떤가용?

```suggestion
interface UserProfileAvatarProps extends Omit<
  AvatarProps,
  "radius" | "variant" | "color" | "fallback"
>{};
```
이것도 강제할 필요까지는 없겠지만 컨벤션 얘기할 때 한 번 얘기하고 가면 좋겠네요 ㅎㅎ

Reply:
- type을 사용한 이유
해당 코리에서 언급했던 내용은, object type의 교차타입에서 type보다 interface가 이점을 가진다는 의미였어용
현 코드에서는 교차 타입이 아닌 Omit 유틸리티 타입을 사용한 타입 변형이기 때문에, type과 interface간 성능 차이가 없고 type으로 간결하게 표현하는 것이 좋다고 생각해서 type을 사용했어요!

- type 선언 방식 컨벤션도 얘기해보는것 좋아요 ㅎㅎ 회의 안건으로 올려놓겠습니당💪🏻

Reply:
아항!

> 공통 컴포넌트는 타입을 확장시켜 사용할 가능성(ex. styled component 등으로 한 번 더 wrapping 해서 사용 가능)이 있기 때문에, interface extends로 prop type을 확장시키는 것이 성능 측면에서 좋다고 생각해요 ~!

저는 이부분을 말씀드린거긴한데,요 측면에서는 어떻게 생각하시는지도 궁금해요! 👀

Reply:
아하! 
* 저는 이 UserProfileAvatar 컴포넌트가 Avatar UI 컴포넌트를 한 번 더 추상화 한 컴포넌트이고, 컴포넌트 맥락이 갖는 특수성이 강해 더 이상 확장해서 사용할 가능성이 적은 컴포넌트라고 생각하여, type으로 정의해도 괜찮다고 판단했어요!
* 참고로 styledComponent의 타입 생성은 내부적으로 type intersection을 사용하고 있기 때문에, 이 경우도 큰 영향이 없을것 같네용
(-> 'UserListItemCard'의 경우 컴포넌트의 prop이 object type의 교차 타입이기 때문에 내부 타입 교차 비용을 줄여야 외부에서 intersection 하는 케이스에서도 비용이 줄어든다고 판단하여 언급했던 내용이었어요~)
<img width="687" alt="image" src="https://github.com/user-attachments/assets/5811264f-71df-4582-8495-8197ab8b776c" />

Reply:
type으로만 표현할수 있는 타입들도 있기 때문에, 개인적인 코드 스타일은
1) 컴포넌트의 prop이 object type의 intersection일 때는 interface로 선언, 단순한 타입 정의나 타입 변환 같은 케이스는 type으로 선언한 뒤,
2) 컴포넌트가 확장되어 사용되는 케이스 (기존 컴포넌트를 상속하여 새로운 컴포넌트를 만들거나, HOC로 사용하거나...)로 사용될 때 interface 형식으로 바꿔주는
기준을 가지고 type vs interface를 사용하고 있는데,

'컴포넌트의 확장 가능성'에 대한 기준이 서로 다를 수도 있을것 같아서 이 부분을 판단하기 애매하다고 생각하면 공통 컴포넌트는 interface로 선언하자로 통일해도 좋을것 같네요!
요 부분을 회의에서 논의해봅시당 !! 저도 장단점을 더 생각해 볼게요 ㅎㅎ

Reply:
저는 `Omit`과 같은 유틸리티 타입을 사용하는 케이스에서는 type alias를 사용하는 게 더 적합하다는 생각이 들어요!!

interface는 객체의 타입을 확장하거나 상속할 때 이점이 있다고 생각해요. 이전 리뷰에서 연서가 제안 준 케이스도 기존에 있던 객체 인터페이스를 확장하는 케이스이기도 했구요!

지금은 사실 상속을 받는 상황이 아니라 특정 타입에서 필요없는 속성들만 제거해서 만든 타입이기도 하고, 연서 말대로 더 이상 확장해서 사용할 가능성이 적은 컴포넌트라는 생각이 들어서 interface + extends보다는 type alias를 사용하는 것이 더 직관적이고 적합하다는 생각이 들어요!

개인적으로 type alias / interface 중 하나만 정해서 모든 케이스에 대해 통일하기보다는, 상황에 따라 적합한 것을 골라 쓰는 게 좋다는 생각이 듭니당 👀 이 부분은 서현이랑 연서 말대로 회의 때 맞추고 가면 좋을 것 같아요!! 😃

Reply:
저도 상황 따라 쓰는거 좋아용 ~~ 머지 고고 하시져!